### ISSUE TITLE
Twisted's test suite fails on Windows when there's spaces in sys.executable

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8046]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8046|
|Type|defect|
|Created|2015-10-10 04:26:10Z|
|Branch|https://github.com/twisted/twisted/tree/windows-spaces-8046|

This is causing trunk to fail right now.

<details><summary>Searchable metadata</summary>

```
trac-id__8046 8046
type__defect defect
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__branches_windows_spaces_8046 branches/windows-spaces-8046
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1444451170767393 1444451170767393
changetime__1444463666549037 1444463666549037
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [45744]) Branching to windows-spaces-8046.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Tests are passing. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set owner to @hawkowl|
|-|-|

LGTM. Very glad to get back to 100% green. Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [45747]) Merge windows-spaces-8046: Make the test suite pass on Windows when sys.executable contains spaces

Author: hawkowl
Reviewer: glyph
Fixes: [#8046](https://github.com/twisted/twisted/issues/8046)
